src/pricecypher/endpoints/config.py

```diff
@@ -0,0 +1,59 @@
+from typing import Optional
+
+from pricecypher.endpoints.base_endpoint import BaseEndpoint
+from pricecypher.exceptions import PriceCypherError
+from pricecypher.models import ConfigSection, ConfigSectionWithKeys
+
+
+class ConfigEndpoint(BaseEndpoint):
+    """PriceCypher config service endpoints.
+
+    :param RestClient client: HTTP client for making API requests.
+    :param int dataset_id: Dataset ID.
+    :param str config_base: (optional) Base URL for PriceCypher config service API.
+        (defaults to https://config.pricecypher.com)
+    """
+
+    def __init__(self, client, dataset_id, config_base='https://config.pricecypher.com'):
+        self.base_url = config_base
+        self.client = client
+        self.dataset_id = dataset_id
+
+    def sections(self):
+        """
+        Sections endpoints in config service API.
+        :rtype: SectionsEndpoint
+        """
+        return SectionsEndpoint(self.client, self._url(['api/datasets', self.dataset_id, '/config/sections']))
+
+
+class SectionsEndpoint(BaseEndpoint):
+    """
+    PriceCypher sections endpoints in config service API.
+    """
```

Comment:
Should the docstring include :param: for init args?

Reply:
This class is not "public", it's only used internally by the ConfigEndpoint class. Note that it's also not exported in the `endpoints` package `__init__.py`. 

IMO we don't need to be very strict with private classes/functions. Unless it's really not clear what is going on or e.g. unexpected parameters are required of course. Personally I don't think that's the case here.